iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -58,16 +71,19 @@ public override async Task SendTelemetryAsync(TelemetryMessage message, Cancella
             if (Logging.IsEnabled)
                 Logging.Enter(this, message, cancellationToken, nameof(SendTelemetryAsync));
 
+            await VerifyIsOpenAsync(cancellationToken).ConfigureAwait(false);
+            using var operationCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _cancelPendingOperationsCts.Token);
```

Comment:
I've added a pre-check operation to each API call - `VerifyIsOpenAsync()` once outside the retry block.

If the verification that the client is open is always linked with the retry block, then closing a client and then calling `SendTelemetryAsync()` will result in the SDK complaining that the linked cancellation token source (the one that signals in-progress operations for closure) has been canceled already, instead of complaining that the client is closed.

Reply:
Updating the formatting marked this comment as outdated. Resolving this and adding a new one.